### ISSUE TITLE
Define otype macros similar to __CHERI_CAP_PERMISSION_*__

### ISSUE THREAD

**Issue by jrtc27** [CTSRD-CHERI/llvm-project#566]
This would remove the only `#ifdef` in cheriintrin.h and provide a way for code that can't use compiler-provided headers (like the FreeBSD kernel or, apparently, musl) to avoid having to hard-code things.

**Comment by arichardson** [comment]
Were you thinking of `__CHERI_CAP_TYPE_SENTRY__`? I guess we could also drop the CAP and use  `__CHERI_TYPE_SENTRY__`.

**Comment by jrtc27** [comment]
Yeah. Dropping the CAP would be nicer but then it'd be inconsistent...
